src/NovelRT.Interop/Audio/NrtAudioService.cpp

```diff
@@ -333,7 +333,105 @@ extern "C"
         }
 
         auto serv = reinterpret_cast<Audio::AudioService*>(service);
-        serv->tearDown();
+        serv->TearDown();
+        return NRT_SUCCESS;
+    }
+
+    NrtBool Nrt_AudioService_IsMusicLoaded(NrtAudioServiceHandle service, NrtAudioServiceIteratorHandle handle)
+    {
+        if (service == nullptr)
+        {
+            Nrt_setErrMsgIsNullptrInternal();
+            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+        }
+
+        auto serv = reinterpret_cast<Audio::AudioService*>(service);
+
+        return serv->IsLoaded(reinterpret_cast<NovelRT::Audio::SoundBank::iterator&>(handle)) ? NRT_TRUE : NRT_FALSE;
+    }
+
+    NrtBool Nrt_AudioService_IsSoundLoaded(NrtAudioServiceHandle service, unsigned int handle)
+    {
+        if (service == nullptr)
+        {
+            Nrt_setErrMsgIsNullptrInternal();
+            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+        }
+
+        auto serv = reinterpret_cast<Audio::AudioService*>(service);
+
+        return serv->IsLoaded(handle) ? NRT_TRUE : NRT_FALSE;
+    }
+
+    NrtBool Nrt_AudioService_IsMusicPlaying(NrtAudioServiceHandle service)
+    {
+        if (service == nullptr)
+        {
+            Nrt_setErrMsgIsNullptrInternal();
+            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+        }
+
+        auto serv = reinterpret_cast<Audio::AudioService*>(service);
+
+        return serv->IsMusicPlaying() ? NRT_TRUE : NRT_FALSE;
+    }
+
+    NrtBool Nrt_AudioService_IsSoundPlaying(NrtAudioServiceHandle service, unsigned int handle)
+    {
+        if (service == nullptr)
+        {
+            Nrt_setErrMsgIsNullptrInternal();
+            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+        }
+
+        auto serv = reinterpret_cast<Audio::AudioService*>(service);
+
+        return serv->IsSoundPlaying(handle) ? NRT_TRUE : NRT_FALSE;
+    }
+
+    NrtResult Nrt_AudioService_GetMusicVolume(NrtAudioServiceHandle service, float* output)
+    {
+        if (service == nullptr)
+        {
+            Nrt_setErrMsgIsNullptrInternal();
+            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+        }
+
+        auto serv = reinterpret_cast<Audio::AudioService*>(service);
+
+        try
+        {
+            *output = serv->GetMusicVolume();
+        }
+        catch (const Exceptions::NotInitialisedException)
+        {
+            Nrt_setErrMsgIsNotInitialisedInternal();
+            return NRT_FAILURE_NOT_INITIALISED;
+        }
+
+        return NRT_SUCCESS;
+    }
```

Comment:
Out of curiosity, is this getter this way because of the exception? I know most getters are straight up "grab the value or segfault"

Reply:
I can adjust this to segfault - I just realized that the audio service most likely wouldn't throw a Not Initialised exception here anyways

Reply:
If you have a look at some of our "high perf" getters, they just straight up return the value or segfault. However, if the getter can throw, then this is 100% correct. :)